### ISSUE TITLE
UnitQuaternion::euler_angles returning NaN

### ISSUE THREAD

**Issue by whatisaphone** [rustsim/nalgebra#494]
I found a case where `euler_angles` is broken:

```rust
let quat = UnitQuaternion::from_quaternion(Quaternion::new(-0.10405792, -0.6993922f32, -0.10406871, 0.69942284));
eprintln!("quat = {:?}", quat);
eprintln!("quat.to_rotation_matrix() = {:?}", quat.to_rotation_matrix());
eprintln!("quat.euler_angles() = {:?}", quat.euler_angles());
```

```
quat = Unit { value: Quaternion { coords: Matrix { data: [-0.6993922, -0.10406871, 0.69942284, -0.10405792] } } }
quat.to_rotation_matrix() = Rotation { matrix: Matrix { data: [-0.000045090914, 0.000008717179, -1.0000001, 0.29113066, -0.9566834, -0.000021457672, -0.9566834, -0.29113066, 0.000040620565] } }
quat.to_euler_angles() = (NaN, NaN, NaN)
```

That's a valid rotation, so there shouldn't be any `NaN`s. I noticed that the rotation matrix has a term of -1.0000001, could that be the problem?

**Comment by sebcrozet** [comment]
Thank you for the report!

> That's a valid rotation, so there shouldn't be any NaNs. I noticed that the rotation matrix has a term of -1.0000001, could that be the problem?

This will indeed cause an issue at [that line](https://github.com/rustsim/nalgebra/blob/master/src/geometry/rotation_specialization.rs#L374) of the conversion.

The issue will be fixed by https://github.com/rustsim/nalgebra/pull/495.

**Comment by sebcrozet** [comment]
The fix is released with v0.16.12.

**Comment by whatisaphone** [comment]
Thanks for the quick fix!
